cpp/src/sort/rank.cu

```diff
@@ -220,15 +224,17 @@ std::unique_ptr<column> rank(column_view const &input, rank_method method,
                              bool percentage,
                              rmm::mr::device_memory_resource *mr,
                              cudaStream_t stream = 0) {
-  std::unique_ptr<column> rank_column = [&_include_nulls, &input, &mr, &stream] {
+  data_type const output_type = (percentage or method == rank_method::AVERAGE)
+                                    ? data_type(FLOAT64)
+                                    : data_type(INT32);
```

Comment:
Don't hard code the type for `size_type` in case we change it.

```suggestion
                                    : data_type(type_to_id<size_type>());
```